src/query/string_opfunc.c

```diff
@@ -3445,7 +3438,28 @@ db_json_keys (DB_VALUE * result, DB_VALUE * arg[], int const num_args)
     }
 
   result_json = db_json_allocate_doc ();
-  error_code = db_json_keys_func (*new_doc, result_json, path.c_str ());
+
+  switch (DB_VALUE_DOMAIN_TYPE (arg[0]))
+    {
+    case DB_TYPE_CHAR:
+    case DB_TYPE_VARCHAR:
+    case DB_TYPE_NCHAR:
+    case DB_TYPE_VARNCHAR:
+      error_code = db_json_keys_func (DB_PULL_STRING (arg[0]), *result_json, path.c_str ());
```

Comment:
i think it'll be more clear if db_json_keys_func allocates memory to result_json instead of the calling function